File: Python Banking System/StudentCheckingAccount.py
from CheckingAccount import CheckingAccount

validSchools = ["Wilfrid Laurier", "University of Waterloo", "University of Toronto", "Toronto Metropolitan University"]

class StudentCheckingAccount(CheckingAccount):
    def __init__(self, accountNumber, accountHolderName, balance, bankType, school):
        super().__init__(accountNumber, accountHolderName, balance, bankType)
        self._school = school

    def atmWithdraw(self, amount, atmType):
        if self._school in validSchools:
            print("-Account type registered with valid school, fees ignored-")
            return super().atmWithdraw(amount, "Exception")
        else:
            return super().atmWithdraw(amount, atmType)

    def displayAccountDetails(self):
        print("Displaying information...")
        print(f"Account Number: {self._accountNumber}")
        print(f"Account Name: {self._accountHolderName}")
        print(f"Account Balance: {self._balance}")
        print(f"Bank Type: {self._bankType}")
        print(f"School: {self._school}")
        print()

File: Java Banking System/demo/src/main/java/com/example/StudentCheckingAccount.java
package com.example;

public class StudentCheckingAccount extends CheckingAccount{
    public static String[] validSchools = {"Wilfrid Laurier", "University of Waterloo", "University of Toronto", "Toronto Metropolitan University"};

    public String school;
    public StudentCheckingAccount(int accountNumber, String accountHolderName, int balance, String bankType, String school) {
        super(accountNumber, accountHolderName, balance, bankType);
        this.school = school;
    }
    
    @Override
    public void atmWithdraw(int amount, String atmType){
        for (String i: validSchools) {
            if (i.compareTo(this.school) == 0) {
                System.out.println("-Account type registered with valid school, fees ignored-");
                super.atmWithdraw(amount, "Exception");
            }
        }
        super.atmWithdraw(amount, atmType);
    }

    @Override
    public void displayAccountDetails(){
        System.out.println("Displaying information...");
        System.out.println("Account Number: " + this.getAccountNumber());
        System.out.println("Account Name: " + this.getAccountHolderName());
        System.out.println("Account Balance: " + this.getBalance());
        System.out.println("Bank Type: " + this.getBankType());
        System.out.println("School: " + this.school + "\n");
    }
}
